app/src/main/kotlin/org/mifos/mobile/cn/ui/mifos/accountsFilter/AccountsFilterBottomSheet.kt

```diff
@@ -0,0 +1,83 @@
+package org.mifos.mobile.cn.ui.mifos.accountsFilter
+
+import android.app.Dialog
+import android.content.Context
+import android.content.DialogInterface
+import android.os.Bundle
+import android.support.design.widget.BottomSheetBehavior
+import android.support.design.widget.BottomSheetDialog
+import android.support.design.widget.BottomSheetDialogFragment
+import android.support.v7.widget.LinearLayoutManager
+import android.view.View
+import android.widget.Toast
+import kotlinx.android.synthetic.main.bottom_sheet_filter_accounts.view.*
+import org.mifos.mobile.cn.R
+import org.mifos.mobile.cn.data.models.CheckboxStatus
+import org.mifos.mobile.cn.enums.AccountType
+import org.mifos.mobile.cn.ui.adapter.CheckboxAdapter
+import org.mifos.mobile.cn.ui.base.MifosBaseActivity
+import org.mifos.mobile.cn.ui.mifos.loanApplication.BottomSheetEvent
+import org.mifos.mobile.cn.ui.utils.DataBus
+import org.mifos.mobile.cn.ui.utils.StatusUtils
+import javax.inject.Inject
+
+/**
+ * @author Manish Kumar
+ * @since 20/July/2018
+ */
+
+class AccountsFilterBottomSheet : BottomSheetDialogFragment(), View.OnClickListener {
+
+    private lateinit var rootView: View
+    private lateinit var behavior: BottomSheetBehavior<*>
+    lateinit var accountType: AccountType
+
+    @Inject
+    lateinit var checkboxAdapter: CheckboxAdapter
+
+    var filterList: List<CheckboxStatus>? = null
+
+
+    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
+        val dialog = super.onCreateDialog(savedInstanceState) as BottomSheetDialog
+        (activity as MifosBaseActivity).activityComponent.inject(this)
+        rootView = View.inflate(context, R.layout.bottom_sheet_filter_accounts, null)
+
+        checkboxAdapter.statusList = filterList!!
+
+        rootView.btn_filter.setOnClickListener(this)
+        val layoutManager = LinearLayoutManager(activity)
+        layoutManager.orientation = LinearLayoutManager.VERTICAL
+        rootView.rv_accounts_filter.layoutManager = layoutManager
+        rootView.rv_accounts_filter.adapter = checkboxAdapter
+
+        dialog.setContentView(rootView)
+        behavior = BottomSheetBehavior.from(rootView.parent as View)
+
+        return dialog
+    }
+
+    override fun onClick(v: View?) {
+        when (v?.id) {
```

Comment:
don't implement onClick, directly use it in onCreateDialog

Reply:
@therajanmaurya  done  in previous PR